test/index.js

```diff
@@ -122,8 +122,16 @@ function runTests (name, compat) {
   var algos = ['sha1', 'sha224', 'sha256', 'sha384', 'sha512']
   algos.forEach(function (algorithm) {
     fixtures.valid.forEach(function (f) {
-      var key = f.key || Buffer.from(f.keyHex, 'hex')
-      var salt = f.salt || Buffer.from(f.saltHex, 'hex')
+      if (f.keyuint8) {
+        var key = new Uint8Array(f.keyuint8)
+      } else {
+        var key = f.key || Buffer.from(f.keyHex, 'hex')
+      }
+      if (f.saltuint8) {
+        var salt = new Uint8Array(f.saltuint8)
+      } else {
+        var salt = f.salt || Buffer.from(f.saltHex, 'hex')
+      }
```

Comment:
I found it hard to understand this code block. Would this version be better? Not sure though

```js
      var key = f.key
        || (f.keyHex && Buffer.from(f.keyHex, 'hex'))
        || (f.keyUint8Array && new Uint8Array(f.keyUint8Array))
      var salt = f.salt
        || (f.saltHex && Buffer.from(f.saltHex, 'hex'))
        || (f.saltUint8Array && new Uint8Array(f.saltUint8Array))
```

Reply:
I don't really think that your version is more readable, but that is just my opinion. I instead replaced the old version with one that at least consistently uses `if` instead of the `||` operator. What do you think?

Reply:
Looks good. This is more symmetric as before. Now all 3 options are handled in one flow. 👍